src/main/java/picocli/CommandLine.java

```diff
@@ -12356,6 +12409,13 @@ private void processArguments(List<CommandLine> parsedCommands,
                 }
 
                 // if we find another command, we are done with the current command
+                if (commandSpec.parser().abbreviatedSubcommandsAllowed()) {
+                    try {
+                        arg = AbbreviationMatcher.match(commandSpec.subcommands().keySet(), arg);
+                    } catch (IllegalArgumentException ex) {
+                        throw new ParameterException(CommandLine.this, "Error: " + ex.getMessage(), ex);
```

Comment:
We can pass the `CommandLine` instance as a parameter to the `match` method so it can throw a `ParameterException` instead of an `IllegalArgException`